cli/integrity/integrity_test.go

```diff
@@ -0,0 +1,97 @@
+package integrity
+
+import (
+	"testing"
+
+	"github.com/spf13/cobra"
+	"github.com/stretchr/testify/require"
+)
+
+func TestNewSigner(t *testing.T) {
+	testCases := []struct {
+		name           string
+		privateKeyPath string
+	}{
+		{
+			name:           "Empty path",
+			privateKeyPath: "",
+		},
+		{
+			name:           "Arbitrary path",
+			privateKeyPath: "private.pem",
+		},
+	}
+
+	for _, testCase := range testCases {
+		t.Run(testCase.name, func(t *testing.T) {
+			_, err := NewSigner(testCase.privateKeyPath)
+			require.Equal(t, err, nil, "did not expect an error")
+		})
+	}
+}
+
+func TestRegisterWithIntegritySigner(t *testing.T) {
+	someStr := ""
+
+	testCases := []struct {
+		name    string
+		command *cobra.Command
+		dst     *string
+	}{
+		{
+			name:    "Empty command and dst",
+			command: nil,
+			dst:     nil,
+		},
+		{
+			name:    "Empty command",
+			command: nil,
+			dst:     &someStr,
+		},
+		{
+			name:    "Empty dst",
+			command: &cobra.Command{},
+			dst:     nil,
+		},
+		{
+			name:    "Nothing empty",
+			command: &cobra.Command{},
+			dst:     &someStr,
+		},
+	}
+
+	for _, testCase := range testCases {
+		// This simply mustn't die.
+		t.Run(testCase.name, func(t *testing.T) {
+			RegisterWithIntegrityFlag(testCase.command, testCase.dst)
```

Comment:
So, what actually do you test here?

Reply:
The only way the function with such signature can die is by dereferencing the null pointer. So, the idea is to throw various combinations of null pointers at it and see if any would trigger a runtime error

Reply:
> The only way the function with such signature can die is by dereferencing the null pointer.

It also could add a flag into a command.